platform/ios/src/MGLMapView.mm

```diff
@@ -5994,9 +5994,14 @@ - (CGPoint)userLocationAnnotationViewCenter
         contentFrame = self.contentFrame;
     }
     CGPoint center = CGPointMake(CGRectGetMidX(contentFrame), CGRectGetMidY(contentFrame));
+    
+    if ([self.delegate respondsToSelector:@selector(mapViewUserLocationAnchorPoint:)])
+    {
+        center = [self.delegate mapViewUserLocationAnchorPoint:self];
```

Comment:
The delegate should be expressing the anchor point in relation to the content frame, so offset the anchor point by the content insets.